liana/method/_Method.py

```diff
@@ -267,3 +277,153 @@ def __call__(self,
+            liana, or any of the columns returned by `scanpy.tl.rank_genes_groups`, each
+            starting with ligand_* or receptor_*. For example, `['ligand_pvals', 'receptor_pvals']`.
+            `None` by default.
+        return_all_lrs
+            Bool whether to return all LRs, or only those that surpass the `expr_prop`
+            threshold. Those interactions that do not pass the `expr_prop` threshold will
+            be assigned to the *worst* score of the ones that do. `False` by default.
+        use_raw
+            Use raw attribute of adata if present.
+        layer
+            Layer in anndata.AnnData.layers to use. If None, use anndata.AnnData.X.
+        de_method
+            Differential expression method. `scanpy.tl.rank_genes_groups` is used to rank genes
+            according to 1vsRest. The default method is 't-test'. Only relevant if p-values
+            are included in `supp_cols`
+        verbose
+            Verbosity flag
+        resource
+            Parameter to enable external resources to be passed. Expects a pandas dataframe
+            with [`ligand`, `receptor`] columns. None by default. If provided will overrule
+            the resource requested via `resource_name`
+        inplace
+            If true return `DataFrame` with results, else assign to `.layer`.
+
+        Returns
+        -------
+            If ``inplace = False``, returns a `DataFrame` with ligand-receptor results
+            Otherwise, modifies the ``adata`` object with the following key:
+            - :attr:`anndata.AnnData.uns` ``['liana_res']`` with the aforementioned DataFrame
+        """
+        if supp_columns is None:
+            supp_columns = []
+
+        ml_res = liana_pipe(adata=adata,
+                        groupby=groupby,
+                        resource_name=resource_name,
+                        resource=None,
+                        met_est_resource_name=met_est_resource_name,
+                        met_est_resource=None,
+                        expr_prop=expr_prop,
+                        est_fun=est_fun,
+                        score_fun=score_fun,
+                        min_cells=min_cells,
+                        supp_columns=supp_columns,
+                        return_all_lrs=return_all_lrs,
+                        base=base,
+                        de_method=None,
+                        verbose=verbose,
+                        use_raw=use_raw,
+                        n_perms=n_perms,
+                        seed=seed,
+                        layer=layer,
+                        _score = self._method, 
+                        pass_mask=pass_mask,
+                        est_only=est_only,
+                        correct_fdr=correct_fdr,
+                        prop_missing_allowed=prop_missing_allowed,
+                        **kwargs
+                        )
+        
+        if inplace:
+            if est_only:
+                if pass_mask:
```

Comment:
what is mask? & why is it important to pass it or not?

Reply:
its the information which genes are in what set. it takes takes to compute it